deploy/legionPipeline.groovy

```diff
@@ -0,0 +1,222 @@
+def installTools(){
+    sh '''
+    sudo rm -rf .venv
+    virtualenv .venv -p $(which python3)
+    cd legion_test
+    ../.venv/bin/python3 setup.py develop
+    '''
+}
+
+def buildDescription(){
+   currentBuild.description = "${params.Profile} ${params.GitBranch}"
+}
+
+def createCluster() {
+    dir('deploy/ansible'){
+        sh 'env'
+        withCredentials([file(credentialsId: params.Profile, variable: 'CREDENTIAL_SECRETS')]) {
+            withAWS(credentials: 'kops') {
+                wrap([$class: 'AnsiColorBuildWrapper', colorMapName: "xterm"]) {
+                    ansiblePlaybook(
+                        playbook: 'create-cluster.yml',
+                        extras: ' --extra-vars "profile=${Profile} skip_kops=${Skip_kops}"',
+                        colorized: true
+                    )
+                }
+            }
+        }
+    }
+}
+
+def terminateCluster() {
+    dir('deploy/ansible'){
+        withAWS(credentials: 'kops') {
+            wrap([$class: 'AnsiColorBuildWrapper', colorMapName: "xterm"]) {
+                ansiblePlaybook(
+                    playbook: 'terminate-cluster.yml',
+                    extras: ' --extra-vars "profile=${Profile}"',
+                    colorized: true
+                )
+            }
+        }
+    }
+}
+
+def deployLegion() {
+    if (params.DeployLegion){
+        dir('deploy/ansible'){
+            withCredentials([file(credentialsId: params.Profile, variable: 'CREDENTIAL_SECRETS')]) {
+                withAWS(credentials: 'kops') {
+                    wrap([$class: 'AnsiColorBuildWrapper', colorMapName: "xterm"]) {
+                        ansiblePlaybook(
+                            playbook: 'deploy-legion.yml',
+                            extras: ' --extra-vars "profile=${Profile} base_version=${BaseVersion}  local_version=${LocalVersion}"',
+                            colorized: true
+                        )
+                    }
+                }
+            }
+        }
+    }
+    else {
+            print("Skipping Legion Deployment")
+    }
+}
+
+def createjenkinsJobs() {
+    if (params.CreateJenkinsTests){
+        def targetBranch = params.GitBranch
+        sh """
+        cd legion_test
+        ../.venv/bin/pip install -r requirements/base.txt
+        ../.venv/bin/pip install -r requirements/test.txt
+        ../.venv/bin/python setup.py develop
+        cd ..
+
+        .venv/bin/create_example_jobs \
+        "https://jenkins.${params.Profile}" \
+        examples \
+        . \
+        "git@github.com:epam/legion.git" \
+        ${targetBranch} \
+        --connection-timeout 600 \
+        --git-root-key "legion-root-key" \
+        --model-host "" \
+        --dynamic-model-prefix "DYNAMIC MODEL"
+        """
+    }
+    else {
+        println('Skipping Jenkins Jobs creation')
+    }
+}
+
+def runRobotTests() {
+            if (params.UseRegressionTests){
```

Comment:
We should move this back-side effect (conclusions) to higher level

Reply:
fixed